tools/datafix/refresh_ids.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+""" Utility to update the datastore key of each Bug to the new format determined by the pre put hook.
+
+    Does this by deleting and reputting each Bug entry.
+"""
+from google.cloud import ndb
+import osv
+
+import argparse
+import json
+import functools
+import time
+import typing
+
+MAX_BATCH_SIZE = 500
+
+
+class DryRunException(Exception):
+  """This exception is raised to cancel a transaction during dry runs"""
+
+
+def get_relevant_ids(verbose: bool) -> list[str]:
+  relevant_ids = []
+
+  query = osv.Bug.query()
+  query.projection = ["db_id"]
+  print(f"Running initial query on {query.kind}...")
+
+  result: typing.Iterable[osv.Bug] = query.iter()
+  counter = 0
+
+  for res in result:
+    counter += 1
+    # Check if the key needs to be updated
+    if res.key.id() != res.db_id:  # type: ignore
+      relevant_ids.append(res.db_id)
+      if verbose:
+        print(res.db_id + ' - ' + res.key.id())  # type: ignore
+
+  print(f"Found {len(relevant_ids)} / {counter} relevant bugs to refresh.")
+  return relevant_ids
+
+
+def reput_bugs(dryrun: bool, verbose: bool) -> None:
```

Comment:
We already have a `reput_bugs.py` file. Could we simply modify it to add the new function?

Reply:
I copied this file from reput_bugs.py, I think the functionality is different enough I didn't want to over complicate the existing file, since that file already works and does what it intends to do. This is mostly for a one off use-case to update the Bug IDs anyway.

Reply:
I've updated the function names to better clarify the difference.